models/hiking.php

```diff
@@ -34,24 +34,26 @@ public function sqlCreate()
 
       $request = "INSERT INTO hiking (name, difficulty, distance, duration, height_difference) VALUES (\"{$this->m_name}\", \"{$this->m_difficulty}\", \"{$this->m_distance}\", \"{$this->m_duration}\", \"{$this->m_height_difference}\")";
 
-        $pdo = new PDO('mysql:host=localhost;dbname=' . $this->m_database_name . ';charset=utf8', "root", "root");
-        $pdo->query($request);
+        $pdo_requete = new Hiking();
```

Comment:
Pourquoi créé-tu une nouvelle instance de la classe `Hiking` pour utiliser ta méthode ?
Vu que tu es déjà dedans tu peux l'utiliser directement en faisant `$this->pdo_connect()`